packages/plugins/response-cache/src/plugin.ts

```diff
@@ -550,7 +553,8 @@ export function useResponseCache<PluginContext extends Record<string, any> = {}>
           return;
         }
 
-        cache.set(cacheKey, result, identifier.values(), finalTtl);
+        const cacheInstance = cacheFactory(onExecuteParams.args.contextValue);
+        cacheInstance.set(cacheKey, result, identifier.values(), finalTtl);
```

Comment:
Why do you instanciate again here ? You should still have access to the `cacheInstance` declared in the upper scope on l.529 ?